### PR TITLE
Shade AWS instrumentation library when using in agent

### LINKED ISSUE #488: Shade aws-sdk from instrumentation-core when used in auto instrumentation
Otherwise those classes could conflict if manual instrumentation also on the application's class path.

Suggestion for shaded package name:

```
io.opentelemetry.auto.instrumentation.awssdk.v2_2.shaded.AwsSdk
```

### PR DESCRIPTION
**Pull request by anuraaga**
Fixes #488

I considered shading in the aws-sdk auto module instead but don't think it's simple to get tests to use the shaded classes in that case. Using an unpublished shaded jar generated by the manual module instead seems to work quite well.

### CODE REVIEW

File: instrumentation-core/aws-sdk/aws-sdk-2.2/aws-sdk-2.2.gradle [hunk 1]
```diff
@@ -48,3 +53,11 @@ dependencies {
   latestDepTestCompile group: 'software.amazon.awssdk', name: 'dynamodb', version: '+'
   latestDepTestCompile group: 'software.amazon.awssdk', name: 'kinesis', version: '+'
 }
+
+shadowJar {
+  archiveClassifier.set('agent')
```
Comment (trask) [inline instrumentation-core/aws-sdk/aws-sdk-2.2/aws-sdk-2.2.gradle:58]: is this the same?
```suggestion
  archiveClassifier = 'agent'
```

File: gradle/publish.gradle [hunk 2]
```diff
@@ -10,7 +10,7 @@ publishing {
   }
   publications {
     maven(MavenPublication) {
-      if (project.tasks.findByName("shadowJar") != null) {
+      if (project.tasks.findByName("shadowJar") != null && !findProperty('noShadowPublish')) {
```
Comment (trask) [inline gradle/publish.gradle:13]: 👍

**Comment by iNikem** [comment]
I think some documentation is absent somewhere. About this code reuse and why this shady business is needed :)

**Comment by iNikem** [comment]
Doc looks good for now, thank you.

Re "reuse": I think it is time to discuss and decide on how auto and manual/assisted instrumentations will look like. Their final package structure, how they communicate with each other etc. And then we can document all that properly in `CONTRIBUTING` doc.

**Comment by anuraaga** [comment]
Agree it's a good time to decide on that. I think the conversation has gotten split over many issues and PRs like #597 #601 #633 But hopefully there's enough information now with these examples

**Comment by iNikem** [comment]
@anuraaga as you have dealt with this the most, can you write a small document, a proposal of how code should be structured, what should use what etc etc? Together with motivation and any other info that could help future developers as well? I think both google doc and a new section in contribution doc are both good options.

**Comment by anuraaga** [comment]
Sure will do!

**Comment by anuraaga** [comment]
Will merge this after #643

**Pull request merged**